--- origsrc/luarocks-3.5.0/src/luarocks/core/sysdetect.lua	2020-12-11 01:03:20.000000000 +0900
+++ src/luarocks-3.5.0/src/luarocks/core/sysdetect.lua
@@ -329,7 +329,7 @@
       rdata_pos = bytes2number(rdata_pos, LITTLE)
       fd:seek("set", rdata_pos)
       local data = fd:read(512)
-      if data:match("cyggcc") then
+      if data:match("cygwin") or data:match("cyggcc") then
          system = "cygwin"
       end
    end
