--- origsrc/luarocks-3.5.0/src/luarocks/core/cfg.lua	2020-12-11 01:03:20.000000000 +0900
+++ src/luarocks-3.5.0/src/luarocks/core/cfg.lua
@@ -399,13 +399,27 @@
    end
 
    if platforms.cygwin then
+      defaults.lua_modules_path = "/share/lua/luarocks/"..lua_version
+      defaults.lib_modules_path = "/lib/lua/luarocks/"..lua_version
+      defaults.rocks_subdir = "/lib/lua/luarocks/rocks-"..lua_version
       defaults.lib_extension = "so" -- can be overridden in the config file for mingw builds
       defaults.arch = "cygwin-"..target_cpu
       defaults.cmake_generator = "Unix Makefiles"
-      defaults.variables.CC = "echo -llua | xargs gcc"
-      defaults.variables.LD = "echo -llua | xargs gcc"
+      defaults.variables.CC = "gcc"
+      defaults.variables.LD = "gcc"
       defaults.variables.LIBFLAG = "-shared"
       defaults.link_lua_explicitly = true
+      defaults.external_deps_patterns = {
+         bin = { "?.exe", "?" },
+         lib = { "lib?.so", "lib?.so.*", "lib?.dll.a", "?.dll.a",
+                 "lib?.a", "cyg?.dll", "lib?.dll", "?.dll", "?.lib" },
+         include = { "?.h" }
+      }
+      defaults.runtime_external_deps_patterns = {
+         bin = { "?.exe", "?" },
+         lib = { "cyg?.dll", "?.dll", "lib?.dll" },
+         include = { "?.h" }
+      }
    end
 
    if platforms.msys then
